lib/jekyll-archives.rb

```diff
@@ -24,10 +24,19 @@ class Archives < Jekyll::Generator
       }.freeze
 
       def initialize(config = {})
-        @config = Utils.deep_merge_hashes(DEFAULTS, config.fetch("jekyll-archives", {}))
+        archives_config = config.fetch("jekyll-archives", {})
+        if archives_config.is_a?(Hash)
+          @config = Utils.deep_merge_hashes(DEFAULTS, archives_config)
+        else
+          @config = nil
+          Jekyll.logger.warn "Archives:", "Expected a hash but got #{archives_config.inspect}"
```

Comment:
Do we need to force our users to care about data structures? What about something like
```suggestion
          Jekyll.logger.warn "Archives:", "The configuration appears to be invalid"
```

Reply:
@mattr- I'm not sure about this suggestion..
When a third-party plugin (e.g. `jekyll-algolia`) erroneously *sets* `config["jekyll-archives"]` to `nil` explicitly, a simple `"The configuration appears to be invalid"` is going to throw the user into more confusion because:
  - First of all, their config file doesn't have a `jekyll-archives` key.
  - Secondly, they have no context as to why the configuration is invalid.
  - they have no backtrace to debug on their own.